File: Leet-Codes/missingTerm.js
var array_dado = [3, 1, 2];
var values_in_array = [];

var n = array_dado.length;
// n = 3

// 1. Lê os elementos do array
for (var elements = 0; elements < n; elements++) {
    if (array_dado[elements] > n) {
        console.log("error, " + array_dado[elements] + " must not be in the array");
    } else {
        values_in_array.push(array_dado[elements]);
    }
}

// 2. Procura qual número no intervalo [0, n] não foi guardado
for (var terms = 0; terms <= n; terms++) {
    if (!values_in_array.includes(terms)) {
        var value = terms;
        console.log("The remaining value of the array is " + value);
    }
}

File: Leet-Codes/missingTerm.py
# Dado um array nums contendo n números distintos no intervalo [0, n], retorne o único número do intervalo que está faltando no array.
# Ex: [3,0,1] → saída: 2; [0,1] → saída: 2 (pois n=2 e faltou o 2).

array_dado = [1,0,3]
values_in_array = []

n = len(array_dado)
# n = 3

for elements in range(n):
    if(array_dado[elements] > n):
        print(f"error, {array_dado[elements]} must not be in the array")
    else:
        values_in_array.append(array_dado[elements])

for terms in range(n + 1):
    if (terms not in values_in_array):
        value = terms
        print(f"The remaining value of the array is {value}")